util/topgen.py

```diff
@@ -1063,6 +1080,9 @@ def insert_ip_attrs(module: ConfigT, params: ParamsT):
         raise SystemExit("There are ipgen modules with multiple instances: "
                          f"{multi_instance_ipgens}")
 
+    if "gpio" in ipgen_instances:
+        instance = ipgen_instances["gpio"][0]
+        insert_ip_attrs(instance, _get_gpio_params(topcfg))
```

Comment:
This might need to be a loop over the instances at some point, but maybe good enough for now.

Reply:
This is correct. If you expand the lines above you'll if there are multiple instances we raise an error. We need to figure out the multiple ipgen story. The trivial case where all instances are equally parameterized is trivial, and they are differently parameterized requires uniquifying each set with the same parameters. Uniquification has made rapid progress since I put this constraint, and now it seems simpler to support multiples. TBD.

Reply:
Yep, I agree. I intentionally did not touch that part since it is not really related to GPIO, and it would be a "one-for-all" fix, supporting the multiple ipgen IPs for all of them.